app/models/adviser.rb

```diff
@@ -48,12 +57,21 @@ def field_order
 
   private
 
+  def check_for_changes
+    @old_firm_id = firm_id_change.first if firm_id_changed?
+  end
+
   def geocode
     if destroyed?
       firm.geocode
     elsif valid?
       GeocodeAdviserJob.perform_later(self)
     end
+
+    if @old_firm_id.present?
+      IndexFirmJob.perform_later(@old_firm_id)
+      @old_firm_id = nil
+    end
```

Comment:
Further, could we put this in a different method with a different callback? Since it's not directly related to geocoding. Or possibly rename this callback to call through to geocode and a method containing this new code.